timelock-impl/src/main/java/com/palantir/atlasdb/timelock/TimelockNamespaces.java

```diff
@@ -79,7 +80,7 @@ public TimeLockServices get(String namespace) {
     }
 
     public TimeLockServices getIgnoringDisabled(String namespace) {
-        return services.computeIfAbsent(namespace, ns -> createNewClient(ns, true));
+        return Optional.ofNullable(services.get(namespace)).orElseGet(() -> createNewClient(namespace, true));
```

Comment:
This solves the correctness issue, but it potentially creates a client on every call and closes any of them. One way to fix it is just to have a separate `ConcurrentMap` for the ignoring services and not reuse the existing normal services.
Alternatively, and maybe better, we can keep the previous behaviour where an explicit disable still kills the existing service (because it does all the locks etc cleanup for us) and let the ignoring version create a new one but make sure that a `get` does a precondition check to verify it's not disabled after calling `services.computeIfAbsent`

Reply:
Ah interesting - yeah, that would work. So we'd create a new service the first time we call `getIgnoringDisabled`, then future `get` calls with throw until the service is re-enabled.